### PR TITLE
feat(package.json): check for v16

### PR DESCRIPTION
**Pull request by jrhender**
Using v18 when running "npm test" is currently failing. Error is "buffer is not defined"
Maybe related to this issue: https://github.com/jestjs/jest/issues/13651

### CODE REVIEW

**Comment by artursudnik** [comment]
Interesting - I am getting the following when executing tests with node@18
```
TypeError: globals.Headers is not a constructor

      at mergeHeaders (node_modules/ky/umd.js:58:18)
      at deepMerge (node_modules/ky/umd.js:94:16)
      at validateAndMerge (node_modules/ky/umd.js:515:10)
      at Proxy.ky.create (node_modules/ky/umd.js:527:45)
      at Object.<anonymous> (node_modules/@digitalbazaar/http-client/main.js:10:23)
```

**Comment by jrhender** [comment]
Ah that is interesting. Maybe it depends on the precise node version used.

**Comment by artursudnik** [comment]
I am using the latest node@18: v18.18.0 and you?

**Comment by artursudnik** [comment]
I am getting the same result with `node@18.0.0`

**Comment by jrhender** [comment]
I was using `node@v18.16.1 `

**Comment by artursudnik** [comment]
wow! With `v18.16.1`  I am getting the same error as you. What a mess...

**Comment by artursudnik** [comment]
So, for v18 versions smaller than 18.2.0 I am getting 
```
TypeError: globals.Headers is not a constructor

      at mergeHeaders (node_modules/ky/umd.js:58:18)
      at deepMerge (node_modules/ky/umd.js:94:16)
      at validateAndMerge (node_modules/ky/umd.js:515:10)
      at Proxy.ky.create (node_modules/ky/umd.js:527:45)
      at Object.<anonymous> (node_modules/@digitalbazaar/http-client/main.js:10:23)
```

for versions 18.2.0 - 18.16.0
```
 ReferenceError: Buffer is not defined

      at Object.<anonymous> (node_modules/node-fetch/src/utils/form-data.js:7:24)
```

and for 18.17.0 - 18.18.0 again:
```
/Users/artur/WebstormProjects/energyweb/iam/elia-energyblocks-vcs/node_modules/ky/umd.js:1
    TypeError: globals.Headers is not a constructor

      at mergeHeaders (node_modules/ky/umd.js:58:18)
      at deepMerge (node_modules/ky/umd.js:94:16)
      at validateAndMerge (node_modules/ky/umd.js:515:10)
      at Proxy.ky.create (node_modules/ky/umd.js:527:45)
      at Object.<anonymous> (node_modules/@digitalbazaar/http-client/main.js:10:23)
```

**Comment by artursudnik** [comment]
upgrading dependencies does not help

**Comment by jrhender** [comment]
Thanks for the testing 👍 . I think then that justifies this PR of requiring v16 for now. I think troubleshooting the root cause(s) and upgrading to v18 isn't worth it for now

**Comment by artursudnik** [comment]
Yes, fixing this can be exhausting. Unfortunately, node@v16 is going not to be maintained soon.

**Comment by artursudnik** [comment]
I would also add to this PR `.nvmrc` file with a `16` value. It helps quickly switch node versions with nvm and IDEs also consider this when running tasks in a project context.

Adding `.npmrc` file with `engine-strict=true` helps to avoid executing scripts with incorrect node versions. I guess that with this, a `check-engine` script is not needed if I understand correctly a motivation to have it.

File: package.json [hunk 1]
```diff
@@ -4,12 +4,17 @@
   "description": "VCs for Elia Exchange of EnergyBlocks",
   "main": "index.js",
   "scripts": {
+    "check-engine": "node -e \"if (!require('semver').satisfies(process.version, require('./package.json').engines.node)) { console.error('Unsupported Node.js version! Please use a version that matches: ' + require('./package.json').engines.node); process.exit(1) }\"",
+    "pretest": "npm run check-engine",
```
Comment (artursudnik) [inline package.json:8]: I think that this is not necessary as npm supports node engine version checking when having a project set as described in [my comment](https://github.com/energywebfoundation/elia-energyblocks-vcs/pull/23#issuecomment-1737752278)

Comment (jrhender) [reply package.json:8]: I added the `.npmrc` file with `engine-strict=true` in this commit: https://github.com/energywebfoundation/elia-energyblocks-vcs/pull/23/commits/ef34863b3f98c8aa88d9296aff670f20c2ca7d09
However, I think this is only checked when running `npm install`, not when running npm scripts. I was still able to run `npm test` when using node v18.

Comment (artursudnik) [reply package.json:8]: ah, you are right. I don't know why I had the impression it is verified for all npm scripts. So nevermind :)

**Review by artursudnik — APPROVED** [review]

**Pull request merged**